Circular imports error when running gui.plottery tests

### What happened? (You can include a screenshot if it helps explain)

When running pytest in `gui/plottery` on main we are getting circular imports error:
For instance:
`pytest -svv tests/ert/unit_tests/gui/plottery/test_ensemble_plot.py`
will produce
```
ImportError while importing test module '/data/workspace/ert/tests/ert/unit_tests/gui/plottery/test_ensemble_plot.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/data/env/py311-std/lib64/python3.11/site-packages/_pytest/python.py:493: in importtestmodule
    mod = import_path(
/data/env/py311-std/lib64/python3.11/site-packages/_pytest/pathlib.py:582: in import_path
    importlib.import_module(module_name)
/usr/lib64/python3.11/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1204: in _gcd_import
    ???
<frozen importlib._bootstrap>:1176: in _find_and_load
    ???
<frozen importlib._bootstrap>:1147: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:690: in _load_unlocked
    ???
/data/env/py311-std/lib64/python3.11/site-packages/_pytest/assertion/rewrite.py:184: in exec_module
    exec(co, module.__dict__)
tests/ert/unit_tests/gui/plottery/test_ensemble_plot.py:7: in <module>
    from ert.gui.plottery import PlotConfig, PlotContext
src/ert/gui/plottery/__init__.py:13: in <module>
    from .plot_context import PlotContext  # noqa
src/ert/gui/plottery/plot_context.py:5: in <module>
    from ert.gui.tools.plot.plot_api import EnsembleObject
src/ert/gui/tools/plot/__init__.py:4: in <module>
    from .plot_tool import PlotTool
src/ert/gui/tools/plot/plot_tool.py:8: in <module>
    from .plot_window import PlotWindow
src/ert/gui/tools/plot/plot_window.py:13: in <module>
    from ert.gui.plottery import PlotConfig, PlotContext
E   ImportError: cannot import name 'PlotContext' from partially initialized module 'ert.gui.plottery' (most likely due to a circular import) (/data/workspace/ert/src/ert/gui/plottery/__init__.py)
```

It might be related to this commit:  https://github.com/equinor/ert/commit/156e202efac9f112f189acbf53270e2fc31955ec


### What did you expect to happen?

Just run pytest...

### steps to reproduce

```Python
`pytest -svv tests/ert/unit_tests/gui/plottery/test_ensemble_plot.py`
```


### Environment where bug has been observed

- [X] python 3.8
- [ ] python 3.9
- [ ] python 3.10
- [X] python 3.11
- [X] python 3.12
- [ ] macosx
- [X] rhel7
- [X] rhel8
- [ ] local queue
- [ ] lsf queue
- [ ] slurm queue
- [ ] openPBS queue
- [ ] used flag `--enable-scheduler`

--- a/src/ert/gui/plottery/plot_config_factory.py
+++ b/src/ert/gui/plottery/plot_config_factory.py
@@ -1,5 +1,11 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
 from ert.gui.plottery import PlotConfig
-from ert.gui.tools.plot.plot_api import PlotApiKeyDefinition
+
+if TYPE_CHECKING:
+    from ert.gui.tools.plot.plot_api import PlotApiKeyDefinition
 
 
 class PlotConfigFactory:

--- a/src/ert/gui/plottery/plot_context.py
+++ b/src/ert/gui/plottery/plot_context.py
@@ -2,12 +2,11 @@ from __future__ import annotations
 
 from typing import TYPE_CHECKING, ClassVar, List, Optional
 
-from ert.gui.tools.plot.plot_api import EnsembleObject
-
 if TYPE_CHECKING:
     from pandas import DataFrame
 
     from ert.gui.plottery import PlotConfig
+    from ert.gui.tools.plot.plot_api import EnsembleObject
 
 
 class PlotContext:

--- a/src/ert/gui/plottery/plots/ensemble.py
+++ b/src/ert/gui/plottery/plots/ensemble.py
@@ -6,7 +6,6 @@ import numpy as np
 import pandas as pd
 
 from ert.gui.plottery.plots.history import plotHistory
-from ert.gui.tools.plot.plot_api import EnsembleObject
 from ert.summary_key_type import is_rate
 
 from .observations import plotObservations
@@ -18,6 +17,7 @@ if TYPE_CHECKING:
     from matplotlib.figure import Figure
 
     from ert.gui.plottery import PlotConfig, PlotContext
+    from ert.gui.tools.plot.plot_api import EnsembleObject
 
 
 class EnsemblePlot:

